ember-core/shared/src/main/scala/org/http4s/ember/core/h2/H2Client.scala

```diff
@@ -313,6 +313,27 @@ private[ember] class H2Client[F[_]](
 
 private[ember] object H2Client {
   private type TinyClient[F[_]] = Request[F] => Resource[F, Response[F]]
+
+  private type ConnectionPool[F[_]] = Map[H2Client.RequestKey, (H2Connection[F], F[Unit])]
+
+  implicit private class PoolOps[F[_]](private val pool: Ref[F, ConnectionPool[F]]) extends AnyVal {
```

Comment:
I'm in favor of doing regular private functions instead of syntax. Because I'm almost sure that `AnyVal` wouldn't help, and the class will be instantiated.

Reply:
Ok. Turned into internal functions of the main method.